.github/workflows/ci.yaml

```diff
@@ -33,11 +33,21 @@ jobs:
       matrix:
         toxenv: ${{fromJson(needs.gentestmatrix.outputs.matrix)}}
     steps:
-    - name: checkout source code
-      uses: actions/checkout@v2
-    - name: Install tox and docker
-      run: sudo apt update && sudo apt install tox
-    - name: Set up Python
-      uses: actions/setup-python@v2
-    - name: Run tox job
-      run: sudo tox -e ${{ matrix.toxenv }}
\ No newline at end of file
+      - name: checkout source code
+        uses: actions/checkout@v2
+      - name: change the location of the docker data dir to cache images
+        run: |
+          mkdir -p ./data-dir
+          cat /etc/docker/daemon.json | jq --arg PWD $(pwd) '. + { "data-root": ($PWD + "/data-dir") }' | sudo tee /etc/docker/daemon.json
+          sudo systemctl restart docker
+      - name: cache docker images
+        uses: actions/cache@v2
+        with:
+          path: ./data-dir/
+          key: ${{ runner.os }}-docker-images
```

Comment:
If cache is shared between two jobs, I think it makes sense to fetch all images in the job "gentestmatrix". 
Then, the parallel jobs can use the cache already set in place.